### ISSUE TITLE
'Param' with $data expression has unexpected _POST results 

### ISSUE THREAD

**Issue by jwealth** [vitalets/x-editable-yii#130]
When using $data expression in params, the $data in CGridview is incorrectly posted.

``` PHP
'id',
[
    'class' => 'editable.EditableColumn',
    'name' => 'end_datetime',
    'editable' => [
        'params'=>[
            'testId'=>'$data->id',
        ],
        'url'=>$this->createUrl('schedule/editableupdate'),
    ],
],
```

The expected $_POST results for params['testId'] is 16; however, $data->id from the final row of the current page is being used instead..  

```
Request Method:POST
Form Data
name:end_datetime
value:01:00:00
pk:16
testid:10  // This should be the same as the pk above
```

Any help would be great!  Thank you.

**Comment by Bhoft** [comment]
Yes this is a bug, if i use the editable Column the params sent to the post were the params of the "last added column" (last row). So the "dynamic" parameters aren't dynamically set at all.
Btw the scenario was always wrong if i have rows with different scenarios (e.g. insert and update scenario).

I found that if i set  

```
[
    'class' => 'editable.EditableColumn',
    'name' => 'end_datetime',
    'editable' => [
        'params'=>[
            'testId'=>'$data->id',
        ],
        'htmlOptions => [
            'data-scenario' => '$data->scenario',
            'data-testId' => '$data->id',
        ],
        'url'=>$this->createUrl('schedule/editableupdate'),
    ],
],
```

That inside the js in the save function the this.options.scenario as well as the this.options.testId is correctly set. But the "submitted" data in this.options.params which is an array with the additional params has the data values from the "last row".  These are also the parameters which where only set "once" in the js call where editable column is enabled. So the commit https://github.com/vitalets/x-editable-yii/commit/c4660a490aa80277c0b52bfc779f07a3fe78982a wasn't working since then for EditableColumns if each row has other params "values".

The problem with those "params" parameter is that is also used to add the CSRF Token. 
And this is added regardles if params is an array or an js function. Because params could also be something like this:

```
    'params' => 'js: function(params) {
                    //originally params contain pk, name and value
                    params.scenario=$(this).data().scenario;
                    params.testId=$(this).data().testId;
                    return params;
   }',
```

But then also the data-attribute values must be added to each link by the HtmlOptions.
And then you will get a problem with the current code because the CSRF Token is added to params which destroys the params if its not an array (so there is a check also needed).

But i think if Params is an array and those htmlOptions data-attribute fields are set those should be then used for posting.

And therefore I have also created a pull request:
https://github.com/vitalets/x-editable-yii/pull/141
